tests/generic.cpp

```diff
@@ -71,16 +71,33 @@ BOOST_AUTO_TEST_CASE(random_decision_vector_test)
     BOOST_CHECK_THROW(random_decision_vector({{1, -inf}, {0, 32}}, r_engine), std::invalid_argument);
     BOOST_CHECK_THROW(random_decision_vector({{1, 2, 3}, {0, 3}}, r_engine), std::invalid_argument);
     BOOST_CHECK_THROW(random_decision_vector({{0, 2, 3}, {1, 4, nan}}, r_engine), std::invalid_argument);
+    BOOST_CHECK_THROW(random_decision_vector({{0, 2, nan}, {1, 4, 4}}, r_engine), std::invalid_argument);
```

Comment:
So I think there is some confusion on how we deal with infinites. 

Status quo:

 * It is allowed to have infinite bounds on the problem.
 * It is not allowed to draw at random a number if the bounds are infinite.

These cases are already tested on the c++ side.

With the addition of the integer part I would not change this behaviour and keep the possibility to have infinite bounds on the problem (also for integers), while preventing to draw random decision vectors from infinite bounds.

The reason for this seemingly asymmetric behaviour is that
 * It makes sense to signal to the optimizer `algo.evolve()` that the problem is unbounded.
 * It makes little sense, if the bounds are infinite, to construct a decision vector drawing randomly from it, better to have the user explicitly init the pop in that case.